pkg/apis/resource/types.go

```diff
@@ -1937,16 +1944,39 @@ type DeviceTaintSelector struct {
 	//
 	// +optional
 	Device *string
+}
```

Comment:
I was wondering whether I should add `ShareID` as another criteria to `DeviceTaintSelector`. It would enable evicting a very specific set of pods, those with access to a particular share of a device.

I decided against it because:
- The use case seems rather weak.
- It is something that isn't possible with taints published by a driver in a ResourceSlice (those are attached to devices, not shares), so it would be inconsistent.

Agreed?

This hadn't been considered in the KEP planning phase yet, it occurred to me later.

cc @sunya-ch